partiql-planner/src/main/kotlin/org/partiql/planner/internal/typer/PlanTyper.kt

```diff
@@ -515,20 +511,31 @@ internal class PlanTyper(
                     // we might improve upon this with some constant folding prior to typing
                     ANY
                 }
-            }.toSet()
-            val finalType = unionOf(pathTypes + toAddTypes).flatten()
-            return rex(finalType.swallowAny(), rexOpPathKey(root, key))
+            }
+            if (elementType.isEmpty()) {
+                handleAlwaysMissing()
+                return rex(ANY, rexOpPathKey(root, key))
+            }
+            // TODO: SwallowAny should happen by default
+            return rex(unionOf(elementType).swallowAny(), rexOpPathKey(root, key))
```

Comment:
perhaps we could get rid of `.swallowAny()` function overall? I think this PR https://github.com/partiql/partiql-lang-kotlin/pull/1395 fixes the flatten call when one of the types is `AnyType`.